storage/src/main/java/com/tencent/rss/storage/handler/impl/HdfsClientReadHandler.java

```diff
@@ -28,6 +39,10 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(HdfsClientReadHandler.class);
 
+  private final Iterator<Entry<String, HdfsFileReader>> readerIterator;
+  private final List<ShuffleDataSegment> shuffleDataSegments = Lists.newArrayList();
+  private String currentPath = "";
```

Comment:
currentPath is a little weird. I feel we should have a new class to record the mapping relation, and remove currentPath.

Reply:
Will add a iterator wrapper class with getKey, next, hasNext.